subxt/src/config/signed_extensions.rs

```diff
@@ -271,18 +284,20 @@ impl<T: Config> ExtrinsicParams<T> for ChargeAssetTxPayment {
 
 impl ExtrinsicParamsEncoder for ChargeAssetTxPayment {
     fn encode_extra_to(&self, v: &mut Vec<u8>) {
-        (self.tip, self.asset_id).encode_to(v);
+        self.encode_to(v);
```

Comment:
Personally I prefer the prev approach and not making the struct `Encode` (it feels like an implementation detail that's better to not expose). Same as below